aeron-client/src/main/c/aeron_fragment_assembler.h

```diff
@@ -87,4 +96,21 @@ inline int aeron_buffer_builder_append(
     return 0;
 }
 
+inline void aeron_buffer_builder_capture_header(aeron_buffer_builder_t *buffer_builder, aeron_header_t *header)
+{
+    buffer_builder->header.initial_term_id = header->initial_term_id;
+    buffer_builder->header.position_bits_to_shift = header->position_bits_to_shift;
+    memcpy(buffer_builder->header.frame, header->frame, sizeof(aeron_data_header_t));
+}
+
+inline aeron_header_t* aeron_buffer_builder_complete_header(aeron_buffer_builder_t *buffer_builder, aeron_header_t *header)
+{
+    aeron_frame_header_t *frame_header = &buffer_builder->header.frame->frame_header;
+    int32_t max_payload_length = frame_header->frame_length - (int32_t)AERON_DATA_HEADER_LENGTH;
+    frame_header->frame_length =
+        (int32_t)aeron_logbuffer_compute_fragmented_length(buffer_builder->limit, max_payload_length);
+    frame_header->flags = (uint8_t)(header->frame->frame_header.flags | AERON_DATA_HEADER_BEGIN_FLAG);
```

Comment:
Should this also be `|` with the flags from the first header as well?  We could end up losing flags from the first header, e.g. EOS flag and any potential flags that were added in the future.  If so we'd need to look at the Java implementation as well.

Reply:
Good point, I'll change both C and Java implementation to OR first and last frame flags.